File: src/c/magma/java/JavaSetCollector.c
#include "../../magma/api/stream/Collector.h"
struct JavaSetCollector<T> implements Collector<T, JavaSet<T>>{
	JavaSet<T> createInitial(){
		return new JavaSet<>();
	}
	JavaSet<T> fold(JavaSet<T> current, T element){
		return current.add(element);
	}
}

File: src/java/magma/java/JavaSetCollector.java
package magma.java;

import magma.api.stream.Collector;

class JavaSetCollector<T> implements Collector<T, JavaSet<T>> {
    @Override
    public JavaSet<T> createInitial() {
        return new JavaSet<>();
    }

    @Override
    public JavaSet<T> fold(JavaSet<T> current, T element) {
        return current.add(element);
    }
}
